tools/snp.sh

```diff
@@ -248,29 +332,50 @@ install_dependencies() {
 }
 
 set_grub_default_snp() {
-  # Get the path to host kernel and the version for setting grub default
-  local host_kernel=$(echo $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/host/debian/linux-image/boot/vmlinuz*"))
-  local host_kernel_version=$(echo "${host_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
-
-  if cat /etc/default/grub | grep "${host_kernel_version}" | grep -v "^#" 2>&1 >/dev/null; then
-    echo -e "Default grub already has SNP [${host_kernel_version}] set"
-    return 0
-  fi
+  if [ "$ID" = "debian" ] || [ "$ID_LIKE" = "debian" ]; then
+    # Get the path to host kernel and the version for setting grub default
+    local host_kernel=$(echo $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/host/debian/linux-image/boot/vmlinuz*"))
+    local host_kernel_version=$(echo "${host_kernel}" | sed "s|.*/boot/vmlinuz-\(.*\)|\1|g")
+
+    if cat /etc/default/grub | grep "${host_kernel_version}" | grep -v "^#" 2>&1 >/dev/null; then
+      echo -e "Default grub already has SNP [${host_kernel_version}] set"
+      return 0
+    fi
 
-  # Retrieve snp menuitem name from grub.cfg
-  local snp_menuitem_name=$(cat /boot/grub/grub.cfg \
-    | grep "menuentry.*${host_kernel_version}" \
-    | grep -v "(recovery mode)" \
-    | grep -o -P "(?<=').*" \
-    | grep -o -P "^[^']*")
+    # Retrieve snp menuitem name from grub.cfg
+    local snp_menuitem_name=$(cat /boot/grub/grub.cfg \
+      | grep "menuentry.*${host_kernel_version}" \
+      | grep -v "(recovery mode)" \
+      | grep -o -P "(?<=').*" \
+      | grep -o -P "^[^']*")
 
-  # Create default grub backup
-  sudo cp /etc/default/grub /etc/default/grub_bkup
-  
-  # Replace grub default with snp menuitem name
-  sudo sed -i -e "s|^\(GRUB_DEFAULT=\).*$|\1\"Advanced options for Ubuntu>${snp_menuitem_name}\"|g" "/etc/default/grub"
+    # Create default grub backup
+    sudo cp /etc/default/grub /etc/default/grub_bkup
+    
+    # Replace grub default with snp menuitem name
+    sudo sed -i -e "s|^\(GRUB_DEFAULT=\).*$|\1\"Advanced options for Ubuntu>${snp_menuitem_name}\"|g" "/etc/default/grub"
+    
+    sudo update-grub
   
-  sudo update-grub
+  else
+    #RHEL
+    # Get the path to host kernel and the version for setting grub default
+    local host_kernel=$(echo $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/kernel-[0-9]*host*"))
+    echo "host_kernel is = ${host_kernel}"
+
+    local host_kernel_version=$(echo "${host_kernel}" | awk -F'-' '{print $2}')
+    local grub_menu=$(sudo grubby --info=ALL | grep title)
+    host_kernel_version="${host_kernel_version//_/-}"    
+    local snp_kernel="vmlinuz-$host_kernel_version"
```

Comment:
Either prepend `host_kernel_version` with "vmlinuz-" or do it directly on 372. No need for a whole new variable `snp_kernel`. Also what does 368 do?